blob/store.go

```diff
@@ -57,10 +80,24 @@ type KV interface {
 
 	// Len reports the number of keys currently in the store.
 	Len(ctx context.Context) (int64, error)
+}
 
-	// Close allows the store to release any resources held open while in use.
-	// If an implementation has nothing to release, it must return nil.
-	Close(context.Context) error
+// Close closes the specified value. If the concrete type of the implementation
+// includes [io.Closer] or exports a Close method with the signature
+//
+//	Close(context.Context) error
+//
+// then Close invokes it and returns the error it reports. Otherwise, Close
+// returns nil.
+func Close(ctx context.Context, s any) error {
```

Comment:
We can't use a `|` constraint here because the interfaces have methods (i.e., they are not pure constraints). The reflection is probably fine, though, a Store will be closed at most once in its lifetime in ordinary usage.

Reply:
Yup, s'fine, any non-empty close will spend plenty of time doing closure things anyway.